pkg/controller/snapshot_controller_base.go

```diff
@@ -230,6 +240,28 @@ func (ctrl *csiSnapshotController) snapshotWorker() {
 			klog.V(2).Infof("error getting snapshot %q from informer: %v", key, err)
 			return false
 		}
+
+		// Not found in informer cache, attempt to get from snapshotsInProgress.
+		inProgressObj, found := ctrl.snapshotsInProgress.Load(key)
+		if found {
+			snapshot, ok := inProgressObj.(*crdv1.VolumeSnapshot)
+			if !ok {
+				klog.Errorf("expected vs, got %+v", inProgressObj)
+				return false
+			}
+
+			newSnapshot, err := ctrl.checkAndUpdateSnapshotClass(snapshot)
+			if err == nil {
+				klog.V(5).Infof("passed checkAndUpdateSnapshotClass for snapshot %q", key)
+				ctrl.updateSnapshot(newSnapshot)
+			}
+
+			ctrl.snapshotQueue.Forget(keyObj)
```

Comment:
I don't think we want to forget and delete in this case.  We should probably follow the whole L230-235 section.  Can we inject the snapshotsInProgressCheck before L230 so they both go through the same codepath?

Reply:
Makes sense, updated to remove the forget/delete and reworked the informer/snapshotsInProgress checks.